src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -0,0 +1,266 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Globalization;
+using System.Text.Encodings.Web;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.OData.Json
+{
+    /// <summary>
+    /// Implement of <see cref="IJsonWriter"/> that is based on
+    /// <see cref="Utf8JsonWriter"/>.
+    /// </summary>
+    internal class ODataUtf8JsonWriter : IJsonWriter, IDisposable

```

Comment:
```suggestion
    internal sealed class ODataUtf8JsonWriter : IJsonWriter, IDisposable

```

The only `virtual` method on this class is necessary *because* it's not `sealed` anyway, seems good to just seal it

Reply:
We might also consider making this class `public`

Reply:
Thanks. I made it `sealed`. And changed the `protected virtual Dispose()` method to `private`.

I didn't want to make this public yet, since it's still new and untested "in the wild". Furthermore, the current `JsonWriter` is also `internal`.